### PR TITLE
Add ProblemReport section to metadata schema

### PR DESCRIPTION
**Pull request by ddanielr**
Adds ProblemReport section to the metadata schema to consolidate metadata table prefixes.
This removes hardcoded prefix strings located in various classes.

### CODE REVIEW

**Review by dlmarion — APPROVED** [review]
LGTM

**Review by ctubbsii — COMMENTED** [review]
I wonder if some of these APIs have a String equivalent, so we don't have to wrap them all with `new Text`. That could be another slight improvement that would go nicely along with this change (if they are equivalent).

**Comment by ddanielr** [comment]
> I wonder if some of these APIs have a String equivalent, so we don't have to wrap them all with `new Text`. That could be another slight improvement that would go nicely along with this change (if they are equivalent).

That's a great idea! I just added the changes to remove the Text wrappers. 
Mutation already had a CharSequence method signature to use. I didn't think we needed to add a method for Strings, but happy to do so if that's desired. 
 
I also had to modify the Encoding util to support a byte array for full removal of the Hadoop Text wrappers.

This change seemed minimal as Encoding was only being used by ProblemReport and its test class and immediately generated a byte array from the Text object.

**Review by Manno15 — APPROVED** [review]

File: core/src/main/java/org/apache/accumulo/core/metadata/schema/MetadataSchema.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/metadata/schema/MetadataSchema.java:409]: This section definition includes the underscore as a mandatory part of the section. That's the most narrowly scoped definition, and completely fine. However, I'm a bit surprised that the section wasn't defined more widely as everything starting with `err`, as in something like:

```java
      new Section(RESERVED_PREFIX + "err", true, RESERVED_PREFIX + "ers", false);
```

I am not suggesting changing it... just noting the distinction between the narrow section definition and the wider definition.

File: server/base/src/main/java/org/apache/accumulo/server/problems/ProblemReports.java
Comment (ctubbsii) [inline server/base/src/main/java/org/apache/accumulo/server/problems/ProblemReports.java:221]: One difference is that the new version has the start key inclusive, and the old range has it exclusive. I don't think this matters, since we don't expect any rows to just include the prefix without a tableId after the underscore, so this isn't an important change. I'm just noting it here as a very slight difference in behavior, in case there's a surprise later, and this comment serves to help debug something.

File: server/base/src/main/java/org/apache/accumulo/server/problems/ProblemReports.java
Comment (ctubbsii) [inline server/base/src/main/java/org/apache/accumulo/server/problems/ProblemReports.java:223]: This can avoid a Text:

```suggestion
                  scanner.setRange(new Range(ProblemSection.getRowPrefix() + table));
```

**Review by ctubbsii — CHANGES_REQUESTED** [review]
One more minor change to eliminate a `new Text`, but after that, this change looks ready to merge.

File: server/base/src/main/java/org/apache/accumulo/server/problems/ProblemReports.java
Comment (ddanielr) [reply server/base/src/main/java/org/apache/accumulo/server/problems/ProblemReports.java]: I found two more instances of `new Text` use in ProblemReports and was able to remove all three and the import `Text` statement.

**Review by ctubbsii — APPROVED** [review]

**Comment by ctubbsii** [comment]
Thanks, @ddanielr . This can be merged once CI checks are finished.

**Pull request merged**
